test/shelf_io_test.dart

```diff
@@ -137,6 +137,22 @@ void main() {
     expect(response.body, 'Hello from /');
   });
 
+  test('multiple headers are received from the client', () async {
+    await _scheduleServer((request) {
+      return Response.ok('Hello from /', headers: {
+        'requested-values': request.headersAll['request-values'],
+        'requested-values-length':
+            request.headersAll['request-values'].length.toString(),
+      });
+    });
+
+    // TODO: update test once `package:http` is able to send multiple header values
```

Comment:
Can't we do it with `dart:io`?

https://api.dart.dev/stable/2.8.4/dart-io/HttpClient/get.html

Not sure if it concatenates the headers. Otherwise, we could do it with a TCP socket, if it's just a `GET` request it might not be that horrible a hack :see_no_evil:

Reply:
Done.